File: laramie-ingresos-publicos-webapi/src/domain/entities/tasa.ts
export default class Tasa {

    id: number;
	codigo: string;
	idTipoTributo: number;
	idCategoriaTasa: number;
	descripcion: string;
	porcentajeDescuento: number;

	constructor(
        id: number = 0,
		codigo: string = "",
		idTipoTributo: number = 0,
		idCategoriaTasa: number = 0,
		descripcion: string = "",
		porcentajeDescuento: number = 0
	)
	{
        this.id = id;
		this.codigo = codigo;
		this.idTipoTributo = idTipoTributo;
		this.idCategoriaTasa = idCategoriaTasa;
		this.descripcion = descripcion;
		this.porcentajeDescuento = porcentajeDescuento;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.codigo = row.codigo ?? "";
		this.idTipoTributo = row.idTipoTributo ?? 0;
		this.idCategoriaTasa = row.idCategoriaTasa ?? 0;
		this.descripcion = row.descripcion ?? "";
		this.porcentajeDescuento = row.porcentajeDescuento ?? 0;
	}

}

File: modules/tesoreria/models/tasa.py
from sqlalchemy import Column, BigInteger, String, Numeric
from sqlalchemy.orm import relationship

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class Tasa(Base):
    __tablename__ = "tesoreria_tasas"

    id = Column(BigInteger, primary_key=True)
    codigo = Column(String(50), nullable=False)
    id_tipo_tributo = Column(BigInteger, nullable=True)
    id_categoria_tasa = Column(BigInteger, nullable=True)
    descripcion = Column(String(250), nullable=False)
    porcentaje_descuento = Column(Numeric(8, 4), nullable=True)

    sub_tasas = relationship("SubTasa", back_populates="tasa", lazy="dynamic")
